apps/s_client.c

```diff
@@ -2218,7 +2219,20 @@ int s_client_main(int argc, char **argv)
                            "Didn't find STARTTLS in server response,"
                            " trying anyway...\n");
             BIO_printf(sbio, "STARTTLS\r\n");
-            BIO_read(sbio, sbuf, BUFSIZZ);
+            mbuf_len = BIO_read(sbio, mbuf, BUFSIZZ);
+            if (mbuf_len < 0) {
+                BIO_printf(bio_err, "BIO_read failed\n");
+                goto end;
+            }
+            mbuf[mbuf_len] = '\0';
+            if (sscanf(mbuf, "%3d ", &numeric) != 1) {
+                BIO_printf(bio_err, "Server does not support STARTTLS.\n");
+                goto shut;
+            }
+            if (numeric != 382) {
```

Comment:
Suggestion is to skip `sscanf != 1` or `mbuf_len<3` *altogether*. Once again, why do we have this code at all? Does anybody read news with openssl s_client? No. This code serves trouble-shooting purposes. I.e. when your normal client doesn't work, you try to figure out what's wrong and resort for s_client, right? Image server returns something that does not start with number or just couple of bytes. Would you, as trouble-shooter, appreciate to see *that* server reply or "Server does not support STARTTLS."? What would server administrator prefer to hear from you?